gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/LocalRegion.java

```diff
@@ -14352,6 +14353,12 @@ private long getEntryOverhead(RegionEntry entry){
     Object key = entry.getRawKey();
     if (key != null) {
       entryOverhead += ReflectionSingleObjectSizer.INSTANCE.sizeof(key);
+    }else{
+      // first key.
+      Object firstKey = this.getRegionMap().keySet().iterator().next();
+      if(firstKey != null){
+        entryOverhead += ReflectionSingleObjectSizer.INSTANCE.sizeof(firstKey);
+      }
```

Comment:
ASFIK key will be part of Gfxd values always. Only a shell object is there to project the key from the value. Only when a value is overflowed to disk we copy the key to the key object. There are separate handling for those situations.